tests/PHPStan/Type/Constant/ConstantStringTypeTest.php

```diff
@@ -148,4 +148,9 @@ public function testGeneralize(): void
 		$this->assertSame('class-string', (new ConstantStringType('NonexistentClass', true))->generalize()->describe(VerbosityLevel::precise()));
 	}
 
+	public function testInvalidEncoding(): void
+	{
+		$this->assertIsString((new ConstantStringType(file_get_contents('invalidUtf8Characters.txt', true)))->describe(VerbosityLevel::value()));
```

Comment:
Is it ok to use file_get_contents in this test or I should move it to a different test?

Reply:
Please produce invalid UTF-8 by using `\x00` etc. characters in the source code directly.